app/src/utils/mapBoundaryUploadHelpers.ts

```diff
@@ -70,6 +81,16 @@ export const parseShapeFile = async (file: File): Promise<Feature[]> => {
   });
 };
 
+/**
+ * Generates a unique ID for a geoJSON Feature.
+ *
+ * @param {Feature} feature
+ * @return {*}  {string}
+ */
+const generateFeatureId = (feature: Feature): string => {
```

Comment:
This was something Alfred and I were playing with.  

The overall problem is that we need a unique key for each feature when rendering them on the map. 

Setting the ID field in the feature is ideal, since then it will be available for all future rendering needs.

We could certainly use a UUID. The only reasoning for the hash was that it would cover the case where the same exact feature was introduced to the UI again (presumably allowing react to optimize that a bit and not re-render it).  But, maybe that scenario isn't super critical and the simplicity/speed of a UUID is preferred.

Thoughts?

Reply:
(Technically `JSON.stringify` is not safe to run on an unknown object, as it doesn't correctly handle circular objects (just crashes eventually when it runs out of memory or something). That being said, its not really something people are to code around since its such a rare issue, and if your code isn't critical/sensitive, it probably won't matter anyways)

Reply:
> The only reasoning for the hash was that it would cover the case where the same exact feature was introduced to the UI again (presumably allowing react to optimize that a bit and not re-render it).

If we want the hash to be the same for two like objects, then `object-hash` would probably be appropriate. The JS doc said "creates a unique hash" which is why I raised the issue of the hashes not being unique.

Reply:
Come to think of it, if two records contain the exact same lat/long, then we probably aren't concerned if one of them doesn't render (due to a matching hash), because you'd only really be able to see one of them anyway. Maybe this is a non-issue?

Reply:
In summary: will use UUID instead of object-hash, because the rendering efficiency gains from hashing aren't efficacious enough to justify the cost of computing the hash